stats/stats.py

```diff
@@ -0,0 +1,324 @@
+            pass
+
+    def get_granulations(self):
+        granulation_arguments = {}
+        for granulation in self.granulations:
+            name = granulation['name']
+            if ((self.queryset.model == Transaction and name in ['created_at', 'updated_at'])
+                or (self.queryset.model == Invoice and name in ['issue_date', 'paid_date'])):
+                granulation_arguments['time_granulation_field'] = name
+                granulation_arguments['time_granulation_interval'] = granulation['value']
+            else:
+                granulation_arguments['additional_granulation_field'] = name
+        return granulation_arguments
+
+    def transactions_count(self, queryset):
+        granulation_arguments = self.get_granulations()
+        modifier_field = 'id'
+        return self.get_data(queryset, modifier_field, **granulation_arguments)
+
+    def documents_count(self, queryset):
+        granulation_arguments = self.get_granulations()
+        modifier_field = 'id'
+        return self.get_data(queryset, modifier_field, **granulation_arguments)
+
+    def transactions_amount_average(self, queryset):
+        granulation_arguments = self.get_granulations()
+        modifier_field = 'amount'
+        return self.get_data(queryset, modifier_field, **granulation_arguments)
+
+    def transactions_amount_total(self, queryset):
+        granulation_arguments = self.get_granulations()
+        modifier_field = 'amount'
+        return self.get_data(queryset, modifier_field=modifier_field, **granulation_arguments)
+
+    def serialize_result(self, queryset):
+        tupleList = []
+
+        additional_granulation = self.get_granulations().get('additional_granulation_field')
+
+        for i in queryset:
+            timestamp_date = date(
+                year=i['year'], month=i.get('month', 1), day=i.get('day', 1)
+            )
+
+            if i.get('week'):
+                timestamp_date += timedelta(days=i['week'] * 7 - 1)
+                print(i['week'], timedelta(days=i['week'] * 7 - 1))
+
+            timestamp = timestamp_date.strftime('%s')
+
+            datapoint = ((timestamp, i[self.MODIFIER_ANNOTATION])
+                         if additional_granulation is None
+                         else (timestamp, i[additional_granulation],
+                               i[self.MODIFIER_ANNOTATION]))
+            tupleList.append(datapoint)
+
+        return tupleList
+
+    def documents_overdue_payments(self, queryset):
+        overdue = {}
+        for i in queryset.filter(due_date__lte=datetime.now()):
+            new = {i.customer: [{'total': i.total, 'paid': i.amount_paid_in_transaction_currency,
+                                 'left': i.amount_to_be_charged_in_transaction_currency}]}
+            overdue[i.customer_id] = new
+        return overdue
+
+    def add_list_item(self, stats_list, plan_name, plan_id, subscription_id, subscription_amount, customer):
```

Comment:
This is only called within the `subscriptions_estimated_income` method. You should rename it and maybe precede it with an underscore. Something like `_add_estimated_income_to_list`.